### PR TITLE
Ports Incursion from BeeStation

### PR DESCRIPTION
**Pull request by ghost**
# Document the changes in your pull request

This [ports incursion](https://github.com/BeeStation/BeeStation-Hornet/pull/2168) - team traitors - antagonist from beestation.  Imagine blood brothers, but not boring, with traitor gear and you don't 100% fail if your teammate is an idiot. E

# Wiki Documentation

Add information about it on traitor page.

# Changelog

:cl:  Author: [PowerfulBacon](https://github.com/PowerfulBacon)
rscadd: Ports Incursion from beestation
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
Fuck i forgot to change the name

**Comment by ghost** [comment]
Bruh

**Comment by ghost** [comment]
Help

File: code/modules/antagonists/incursion/incursion.dm [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+	if(!LAZYLEN(people))
+		log_game("Not enough players for incursion role. [LAZYLEN(people)]")
+		return
+	var/datum/mind/target = SSticker.mode.antag_pick(people, ROLE_TRAITOR)
+	if(!target)
+		log_game("No mind selected.")
+		return
+	target.make_Traitor()
+	to_chat(target, "<span class='userdanger'>You have been declared an ex-communicate of the syndicate and are being hunted down.</span>")
+	to_chat(target, "<span class='warning'>You have stolen syndicate objective documents, complete the objectives to throw off the syndicate and sabotage their efforts.</span>")
```
Comment (tattax) [inline code/modules/antagonists/incursion/incursion.dm]:
```suggestion
	to_chat(target, span_warning("You have stolen syndicate objective documents, complete the objectives to throw off the syndicate and sabotage their efforts."))
```

File: code/modules/antagonists/incursion/incursion.dm [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+	var/list/datum/mind/people = SSticker.mode.get_alive_non_antagonsist_players_for_role(ROLE_TRAITOR)
+	if(!LAZYLEN(people))
+		log_game("Not enough players for incursion role. [LAZYLEN(people)]")
+		return
+	var/datum/mind/target = SSticker.mode.antag_pick(people, ROLE_TRAITOR)
+	if(!target)
+		log_game("No mind selected.")
+		return
+	target.make_Traitor()
+	to_chat(target, "<span class='userdanger'>You have been declared an ex-communicate of the syndicate and are being hunted down.</span>")
```
Comment (tattax) [inline code/modules/antagonists/incursion/incursion.dm]:
```suggestion
	to_chat(target, span_userdanger("You have been declared an ex-communicate of the syndicate and are being hunted down."))
```

File: code/modules/antagonists/incursion/incursion.dm [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+		var/datum/uplink_purchase_log/H = GLOB.uplink_purchase_logs_by_key[syndicate.key]
+		if(H)
+			TC_uses += H.total_spent
+			purchases += H.generate_render(show_key = FALSE)
+	parts += "(Syndicates used [TC_uses] TC) [purchases]"
+
+	if(win)
+		parts += "<span class='greentext'>The Syndicate were successful with their operation!</span>"
+	else
+		parts += "<span class='redtext'>The Syndicate failed their incursion!</span>"
```
Comment (tattax) [inline code/modules/antagonists/incursion/incursion.dm]:
```suggestion
		parts += span_redtext("The Syndicate failed their incursion!")
```

File: code/modules/antagonists/incursion/incursion.dm [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+	for(var/I in members)
+		var/datum/mind/syndicate = I
+		var/datum/uplink_purchase_log/H = GLOB.uplink_purchase_logs_by_key[syndicate.key]
+		if(H)
+			TC_uses += H.total_spent
+			purchases += H.generate_render(show_key = FALSE)
+	parts += "(Syndicates used [TC_uses] TC) [purchases]"
+
+	if(win)
+		parts += "<span class='greentext'>The Syndicate were successful with their operation!</span>"
```
Comment (tattax) [inline code/modules/antagonists/incursion/incursion.dm]:
```suggestion
		parts += span_greentext("The Syndicate were successful with their operation!")
```

File: code/modules/antagonists/incursion/incursion.dm [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+	name = "syndicate incursion force"
+	member_name = "incursion member"
+
+/datum/team/incursion/is_solo()
+	return FALSE
+
+/datum/team/incursion/roundend_report()
+	var/list/parts = list()
+
+	parts += "<span class='header'>The members of the Syndicate incursion were:</span>"
```
Comment (tattax) [inline code/modules/antagonists/incursion/incursion.dm]:
```suggestion
	parts += span_header("The members of the Syndicate incursion were:")
```

File: code/modules/antagonists/incursion/incursion.dm [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+		var/datum/mind/M = members[i]
+		member_text += M.name
+		if(i == members.len - 1)
+			member_text += " and "
+		else if(i != members.len)
+			member_text += ", "
+	return member_text
+
+/datum/antagonist/incursion/greet()
+	to_chat(owner.current, "<span class='alertsyndie'>You are the member of a Syndicate incursion team!</span>")
```
Comment (tattax) [inline code/modules/antagonists/incursion/incursion.dm]:
```suggestion
	to_chat(owner.current, span_alertsyndie("You are the member of a Syndicate incursion team!"))
```

File: code/modules/antagonists/incursion/incursion.dm [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+	for(var/datum/objective/O in team.objectives)
+		objectives += O
+	owner.special_role = special_role
+	finalize_incursion()
+	return ..()
+
+/datum/antagonist/incursion/on_removal()
+	SSticker.mode.incursionists -= owner
+	if(owner.current)
+		to_chat(owner.current,"<span class='userdanger'>The Syndicate plans have fallen apart, you are no longer a member of the incursion.</span>")
```
Comment (tattax) [inline code/modules/antagonists/incursion/incursion.dm]:
```suggestion
		to_chat(owner.current, span_userdanger("The Syndicate plans have fallen apart, you are no longer a member of the incursion."))
```

File: code/game/objects/items/implants/implant_misc.dm [hunk 8]
```diff
@@ -153,3 +153,27 @@
 	name = "implanter (internal syndicate radio)"
 	imp_type = /obj/item/implant/radio/syndicate
 
+/obj/item/implant/radio/syndicate/selfdestruct
+	name = "hacked internal radio implant"
+
+/obj/item/implant/radio/syndicate/selfdestruct/on_implanted(mob/living/user)
+	if(!user.mind.has_antag_datum(/datum/antagonist/incursion))
+		user.visible_message("<span class='warning'>[imp_in] starts beeping ominously!</span>", "<span class='userdanger'>You have a sudden feeling of dread. The implant is rigged to explode!</span>")
```
Comment (tattax) [inline code/game/objects/items/implants/implant_misc.dm]:
```suggestion
		user.visible_message(span_warning("[imp_in] starts beeping ominously!"), span_userdanger("You have a sudden feeling of dread. The implant is rigged to explode!"))
```

**Comment by JamieD1** [comment]
Re-open when fixed

**Pull request closed without merging**